src/config/index.ts

```diff
@@ -24,9 +18,31 @@ export const getJSONEnv = (key: string) => {
   return JSON.parse(value);
 };
 
-const config: Config = {
-  appEnv: getEnv('APP_ENV'),
+const configObject = {
+  appEnv: process.env.APP_ENV,
+  serverPort: Number(process.env.PORT || process.env.SERVER_PORT),
   lastCommitHash: getCommitHash()
 };
 
+let cache: Config | undefined;
+function validateEnvs() {
+  try {
+    if (!cache) {
+      cache = configSchema.parse(configObject);
+    }
+  } catch (error) {
+    if (error instanceof ZodError) {
+      error.errors.forEach((validationError) => {
+        logger.error(validationError.message);
+        logger.error(validationError.path.join('.'));
+      });
+    }
+    throw 'ENV Error!';
+  }
+
+  return cache;
+}
```

Comment:
Prefer early return for readablity. plus you can also use `schema.safeParse()` to avoid try catch